app/tasks/email_notifications.rb

```diff
@@ -71,10 +71,22 @@ def feedback_ending(time_text)
 
   private def send_emails(**hash)
     hash[:users].pluck(:email).each_slice(200) do |arr|
+      link = unsubscribe_url token: '#{generate_token(arr.email)}'+'#{arr.email}'
       Mailgun.send_message contest: @contest, text: hash[:text],
                            subject: hash[:subject], bcc_array: arr,
                            to: EMAIL_SINK
     end
     Ajat.info "send_email|#{hash[:subject]}"
   end
+
+  private def generate_token(email)
```

Comment:
1. Would moving this to a method of `UserNotification` be more appropriate?
2. Would you need to add a new column to the `UserNotification` model, and hence, a migration?

Reply:
right